File: 30_days_of_code/python/day_16.py
n = int(input())
a = input().strip().split()
arr = [int(x) for x in arr]
sorted_arr = sorted(arr)
hash = {}
mindiff_arr = []
for i in range(len(sorted_arr)-1):
	mindiff_arr.append(abs(sorted_arr[i] - sorted_arr[i+1]))

mindiff = min(mindiff_arr)

for i in range(len(sorted_arr)):
	hash(sorted_arr[i]) = i

for i in range(len(sorted_arr)):
	if((sorted_arr[i]-mindiff) in hash.keys()):
		print(str(sorted_arr[i]-mindiff)+" "+str(sorted_arr[i])+" ", end="")

File: 30_days_of_code/ruby/day_16.rb
n = gets.strip().to_i
arr = gets.strip().split().map(&:to_i)
sorted = arr.sort()

hash = Hash.new(nil)
mindiff_arr = []

for i in 0...arr.length-1
    mindiff_arr << (sorted[i]-sorted[i+1]).abs
end
mindiff = mindiff_arr.min

for i in 0...arr.length
    hash[sorted[i]] = i
end

for i in 0...arr.length
    if hash.has_key?(sorted[i]-mindiff)
        print (sorted[i]-mindiff).to_s + " " + sorted[i].to_s + " " 
    end
end

